libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1337,6 +1337,20 @@ void ExpressionCompiler::appendArithmeticOperatorCode(Token::Value _operator, Ty
 		m_context << Instruction::DUP2 << Instruction::ISZERO;
 		m_context.appendConditionalJumpTo(m_context.errorTag());
 
+		// Test for (minInt) / (-1)
+		if (c_isSigned && _operator == Token::Div)
+		{
+			m_context << Instruction::DUP1 << Instruction::NOT  // ~x
```

Comment:
Not sure if this is faster, but you might want to use the new inline assembly here:
```
m_context.appendInlineAssembly(R"(
  and(
    eq(x, exp(2, 255)), // x == 2**256 (min int)
    iszero(not(y)) // y == -1
  )
)",
// names of variables already on the stack
{"y", "x"});
m_context.appendConditionalJumpTo(m_context.errorTag());
```

Oh by the way, in all other comments of the code generator, the stack grows to the right, I think yours grows to the left.